src/core/DatePicker/DatePicker.tsx

```diff
@@ -241,9 +252,11 @@ export const DatePicker = (props: DatePickerProps): JSX.Element => {
     if (day.getMonth() !== selectedDay?.getMonth()) {
       setMonthAndYear(day.getMonth(), day.getFullYear());
     }
-    setSelectedDay(day);
-    setFocusedDay(day);
-    onChange?.(day);
+    const newDate = new Date(selectedDay ?? new Date());
+    newDate.setDate(day.getDate());
+    setSelectedDay(newDate);
```

Comment:
Something is fucked up because when selecting a day in next month, it selects the same day but in the current month.

Reply:
yeah, I thought I fixed it.. Might have undone it for other issues...
Will double check..

Reply:
yeah, fixed...